### PR TITLE
Explicitly set github repo in instance presenter spec

### PR DESCRIPTION
**Pull request by mjankowski**
Alternate approach to https://github.com/mastodon/mastodon/pull/24027

### CODE REVIEW

File: spec/presenters/instance_presenter_spec.rb [hunk 1]
```diff
@@ -89,8 +89,28 @@
   end
 
   describe '#source_url' do
-    it 'returns "https://github.com/mastodon/mastodon"' do
-      expect(instance_presenter.source_url).to eq('https://github.com/mastodon/mastodon')
+    context 'with an env variable set' do
```
Comment (nschonni) [inline spec/presenters/instance_presenter_spec.rb:92]:
```suggestion
    context 'with GITHUB_REPOSITORY env variable set' do
```

File: spec/presenters/instance_presenter_spec.rb [hunk 2]
```diff
@@ -89,8 +89,28 @@
...
+          example.run
+        end
+      end
+
+      it 'uses env variable to build a repo URL' do
+        expect(instance_presenter.source_url).to eq('https://github.com/other/repo')
+      end
+    end
+
+    context 'without an env variable set' do
```
Comment (nschonni) [inline spec/presenters/instance_presenter_spec.rb:104]:
```suggestion
    context 'without GITHUB_REPOSITORY env variable set' do
```

**Review by nschonni — APPROVED** [review]

File: Gemfile [hunk 3]
```diff
@@ -117,6 +117,7 @@ end
 
 group :test do
   gem 'capybara', '~> 3.38'
+  gem 'climate_control'
```
Comment (nschonni) [inline Gemfile:120]: Might be worth adding a little description, even though this hasn't been done elsewhere yet
```suggestion

  # Used to mock ENV values in specs
  gem 'climate_control'

```

**Review by ClearlyClaire — APPROVED** [review]

**Comment by bocan** [comment]
I wish "GITHUB_REPOSITORY" had been GIT_REPOSITORY instead, and held the full URL instead of appending to "https://github.com/".  #24027 is one problem, but it would have been nice to point the "View Source Code" link to my Gitlab instance where I build this thing rather than Github where I only use the fork tools.

**Comment by mjankowski** [comment]
> it would have been nice to point the "View Source Code" link to my Gitlab instance..

I didn't actually that closely at this when I was fixing the "github actions breaks on forks" thing ... but it looks like you can set `SOURCE_BASE_URL` env var and it will override the full value.

Separately -- now that I actually look at this, a good spec refactor here would be to move basically all of this stuff out of the instance presenter spec and into a spec for `Mastodon::Version` since thats where the bulk of this logic happens (the presenter just calls a method on the version class).

**Pull request merged**